# Support Vector Regressor (SVR)

In [1]:
# 📌 Preparación de datos
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

#Carga de datos
from google.colab import files
uploaded = files.upload()
for fn in uploaded.keys():
    name=fn
df = pd.read_csv(name, sep=";", encoding='utf-8')

Saving Anexo ET_demo_round_traces_2022.csv to Anexo ET_demo_round_traces_2022.csv


<ipython-input-1-3ecd94279ebf>:14: DtypeWarning: Columns (6,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(name, sep=";", encoding='utf-8')


In [2]:
# Codificación de variable categórica
le = LabelEncoder()
df['Team_encoded'] = le.fit_transform(df['Team'])

# Conversión a numérico
df['MatchKills'] = pd.to_numeric(df['MatchKills'], errors='coerce')
df['MatchAssists'] = pd.to_numeric(df['MatchAssists'], errors='coerce')
df['MatchHeadshots'] = pd.to_numeric(df['MatchHeadshots'], errors='coerce')
df['RoundKills'] = pd.to_numeric(df['RoundKills'], errors='coerce')

# Limpieza de datos nulos
df_clean = df.dropna(subset=['MatchKills', 'MatchAssists', 'MatchHeadshots', 'RoundKills', 'Team'])

# Variables
X = df_clean[['MatchAssists', 'MatchHeadshots', 'RoundKills', 'Team_encoded']]
y = df_clean['MatchKills']

from sklearn.svm import SVR
from sklearn.preprocessing import StandardScaler

# Escalado de datos
scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)

# División de datos
X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size=0.2, random_state=42)

# Modelo
model = SVR()
model.fit(X_train, y_train)
y_pred = model.predict(X_test)

# Métricas
print("💻 Support Vector Regressor")
print(f"R²: {r2_score(y_test, y_pred):.3f}")
print(f"MAE: {mean_absolute_error(y_test, y_pred):.2f}")
print(f"MSE: {mean_squared_error(y_test, y_pred):.2f}")
print(f"RMSE: {np.sqrt(mean_squared_error(y_test, y_pred)):.2f}")

💻 Support Vector Regressor
R²: 0.731
MAE: 2.30
MSE: 10.29
RMSE: 3.21


# 💻 Support Vector Regressor (SVR)

El SVR se comportó bien, con un R² de 0.731 y el MAE más bajo de 2.30, lo que significa que en promedio cometió menos errores que los otros modelos.

Sin embargo, su MSE fue de 10.29 y el RMSE de 3.21, lo que sugiere que aunque fue preciso en muchos casos, falló más cuando los jugadores tenían rendimientos muy altos o muy bajos (partidas sobresalientes o muy malas).

Funciona bien para predecir a jugadores promedio, pero no tanto para casos extremos